opensrp-app/src/main/java/org/smartregister/service/HTTPAgent.java

```diff
@@ -258,74 +202,174 @@ public LoginResponse urlCanBeAccessWithGivenCredentials(String requestURL, Strin
-                entity.addPart("file-category", new StringBody(
-                        image.getFilecategory() != null ? image.getFilecategory() : "profilepic"));
-                ContentBody cbFile = new FileBody(uploadFile, "image/jpeg");
-                entity.addPart("file", cbFile);
-                httpost.setEntity(entity);
-                httpResponse = httpClient.postContent(httpost);
-                responseString = EntityUtils.toString(httpResponse.getEntity());
-                Log.v("response so many", responseString);
-
-                //TODO if response status is not 200 or 201 ?
-                /*
-                int RESPONSE_OK = 200;
-                int RESPONSE_OK_ = 201;
-                if (httpResponse.getStatusLine().getStatusCode() != RESPONSE_OK_
-                        && httpResponse.getStatusLine().getStatusCode() != RESPONSE_OK) {
+            HttpURLConnection httpUrlConnection = initializeHttp(urlString, true);
+
+            httpUrlConnection.setUseCaches(false);
+            httpUrlConnection.setDoInput(true);
+            httpUrlConnection.setDoOutput(true);
+            httpUrlConnection.setRequestProperty("Content-Type", "multipart/form-data;boundary=" + boundary);
+
+            outputStream = httpUrlConnection.getOutputStream();
+            writer = new PrintWriter(new OutputStreamWriter(outputStream, "UTF-8"),true);
+
+            // attach image
+            File uploadImageFile = new File(image.getFilepath());
+            String fileName = uploadImageFile.getName();
+
+            writer.append(twoHyphens + boundary).append(crlf);
+            writer.append("Content-Disposition: form-data; name=\"file\"; filename=\"" + fileName + "\"").append(crlf);
+            writer.append("Content-Type: " + URLConnection.guessContentTypeFromName(fileName)).append(crlf);
+            writer.append("Content-Transfer-Encoding: binary").append(crlf);
+            writer.append(crlf);
+            writer.flush();
+
+            FileInputStream inputStream = new FileInputStream(uploadImageFile);
+            byte[] buffer = new byte[4096];
+            int bytesRead = -1;
+            while ((bytesRead = inputStream.read(buffer)) != -1) {
+                outputStream.write(buffer, 0, bytesRead);
+            }
+            outputStream.flush();
+            inputStream.close();
+
+            writer.append(crlf);
+            writer.flush();
+            // end of attach image
+
+            // adding string params
+            writer.append(twoHyphens + boundary).append(crlf);
+            writer.append("Content-Disposition: form-data; name=\"anm-id\"").append(crlf);
+            writer.append("Content-Type: text/plain; charset=" + "UTF-8").append(crlf);
+            writer.append(crlf);
+            writer.append(image.getAnmId()).append(crlf);
+            writer.flush();
+
+            writer.append(twoHyphens + boundary).append(crlf);
```

Comment:
Create a function for adding the params. Better yet create a MultiPartForm Helper 
e.g on like documented [here](https://stackoverflow.com/a/34409142)

Reply:
My understanding was that fetch uses basic auth and fetchWithCredentials uses username&password authentication. If they are the same, I will just call fetch in fetchWithCredentials